tilequeue/query/rawr.py

```diff
@@ -0,0 +1,578 @@
+from collections import namedtuple, defaultdict
+from shapely.geometry import box
+from shapely.wkb import loads as wkb_loads
+from tilequeue.query.common import layer_properties
+from tilequeue.query.common import is_station_or_stop
+from tilequeue.query.common import is_station_or_line
+from tilequeue.query.common import deassoc
+from tilequeue.query.common import mz_is_interesting_transit_relation
+from tilequeue.query.common import shape_type_lookup
+from tilequeue.transform import calculate_padded_bounds
+from raw_tiles.tile import shape_tile_coverage
+from math import floor
+
+
+class Relation(object):
+    """
+    Relation object holds data about a relation and provides a nicer interface
+    than the raw tuple by turning the tags array into a dict, and separating
+    out the "parts" array of IDs into separate lists for nodes, ways and other
+    relations.
+    """
+
+    def __init__(self, rel_id, way_off, rel_off, parts, members, tags):
+        self.id = rel_id
+        self.tags = deassoc(tags)
+        self.node_ids = parts[0:way_off]
+        self.way_ids = parts[way_off:rel_off]
+        self.rel_ids = parts[rel_off:]
+
+
+class TilePyramid(namedtuple('TilePyramid', 'z x y max_z')):
+    """
+    Represents a "tile pyramid" of all tiles which are geographically
+    contained within the tile `z/x/y` up to a maximum zoom of `max_z`. This is
+    the set of tiles corresponding to one RAWR tile.
+    """
+
+    def tile(self):
+        from raw_tiles.tile import Tile
+        return Tile(self.z, self.x, self.y)
+
+    def bounds(self):
+        from ModestMaps.Core import Coordinate
+        from tilequeue.tile import coord_to_mercator_bounds
+
+        coord = Coordinate(zoom=self.z, column=self.x, row=self.y)
+        bounds = coord_to_mercator_bounds(coord)
+
+        return bounds
+
+    def bbox(self):
+        return box(*self.bounds())
+
+
+# weak enum type used to represent shape type, rather than use a string.
+# (or, where we have to use a string, at least use a shared single instance
+# of a string)
+class ShapeType(object):
+    point = 1
+    line = 2
+    polygon = 3
+
+    _LOOKUP = ['point', 'line', 'polygon']
+
+    @staticmethod
+    def lookup(typ):
+        "turn the enum into a string"
+        return ShapeType._LOOKUP[typ-1]
```

Comment:
What do you think of using the [enum package](https://pypi.python.org/pypi/enum34) here?

Reply:
Good point. At first, it didn't seem necessary to bring that in for something which, at least initially, was very simple. After adding the `lookup` function, it's starting to look like it would be cleaner with the `enum` package. Fixed in #247.